### DISCUSSION TITLE
功能需求，文件加密，多线程分片的上传和下载

### DISCUSSION THREAD

**Discussion by youtehub** [AlistGo/alist#3695] [Ideas]
在 alist 群里面发现好多群友反馈，上传的视频文件，照片等资料，总是被云盘服务器和谐，或者删除，例如百度网盘，阿里网盘，迅雷网盘等，因为国家有限制，某些违规文件不允许保留在网盘内，或者那种设计版权的文件不支持保存等。
所有产生下列需求，在使用 alist 上传文件到网盘的时候，支持使用 **非对称加密等高级加密算法**  实时对文件进行加密，反正就是要是有复杂的加密算法对文件进行加密，在页面开启设置，彻底解决任何文件通过  alist 上传到 网盘服务器后，文件时不时被和谐的问题。
这个需求大部分使用 alist 的人都会遇到，只是以前没有人整理成需求在官方论坛提出来，应该可以使用 golang 实现文件的实时加解密。

百科解释：[实时加解密](https://baike.baidu.com/item/%E5%AE%9E%E6%97%B6%E5%8A%A0%E8%A7%A3%E5%AF%86/502937)

**Comment by youtehub** [comment]
需求之二，多线程，分片下或，面对像夸克网盘，迅雷网盘，之类的限速网盘，目前从用户层面是很难实现提速下或，但是又想使用，所有便产生了第二个需求，多线程，分片下载，golang肯定也能实现，而且由官方来实现更好。
我自己现在就在使用对象存储加中间文件的形式来实现多线程，分片下载，只是现在没有时间学习go语言，所有现在是使用java语言来实现
多线程分片下载，解决某些网盘下载限速的问题，rclone倒是可以设置多线程，rclone也是go语言开发的，不过目前vfs言方似乎还不支持多线程下载，只能等待 rclone的开发者 解决，如果这个需求能实现的话，使用网盘挂载到本地后，在线看 **30G左右的蓝光电影** 都可以。
最好是同时支持，多线程，或者叫多协程，并发，分片，上传，下载
方案：[golang多协程分块下载文件](https://blog.csdn.net/wyfalj/article/details/103582676)
方案：[goLang 实现并发切片上传和下载文件](https://blog.csdn.net/JineD/article/details/125872088)

**Comment by youtehub** [comment]
建议 设置成按钮的形式，让用户自己控制，是否需要启用，多线程分片上传下载，加密上传下载

**Comment by yanjunhui** [comment]
alist只是挂载网盘.....实际下载文件的地址都是各个网盘提供的地址 
是否支持多线程与alist无关的 要看各个网盘官方是否支持
